src/components/AddPlaidBankAccount.js

```diff
@@ -181,52 +181,62 @@ function AddPlaidBankAccount({
-                            if (event === 'ERROR') {
-                                Log.hmmm('[PlaidLink] Error: ', metadata);
-                                if (bankAccountID && metadata && metadata.error_code) {
-                                    BankAccounts.handlePlaidError(bankAccountID, metadata.error_code, metadata.error_message, metadata.request_id);
-                                }
+    const renderPlaidLink = () => {
+        if (Boolean(token) && !bankName) {
+            return (
+                <PlaidLink
+                    token={token}
+                    onSuccess={({publicToken, metadata}) => {
+                        Log.info('[PlaidLink] Success!');
+                        BankAccounts.openPlaidBankAccountSelector(publicToken, metadata.institution.name, allowDebit, bankAccountID);
+                    }}
+                    onError={(error) => {
+                        Log.hmmm('[PlaidLink] Error: ', error.message);
+                    }}
+                    onEvent={(event, metadata) => {
+                        BankAccounts.setPlaidEvent(event);
+                        // Handle Plaid login errors (will potentially reset plaid token and item depending on the error)
+                        if (event === 'ERROR') {
+                            Log.hmmm('[PlaidLink] Error: ', metadata);
+                            if (bankAccountID && metadata && metadata.error_code) {
+                                BankAccounts.handlePlaidError(bankAccountID, metadata.error_code, metadata.error_message, metadata.request_id);
                             }
+                        }
 
-                            // Limit the number of times a user can submit Plaid credentials
-                            if (event === 'SUBMIT_CREDENTIALS') {
-                                App.handleRestrictedEvent(event);
-                            }
-                        }}
-                        // User prematurely exited the Plaid flow
-                        // eslint-disable-next-line react/jsx-props-no-multi-spaces
-                        onExit={onExitPlaid}
-                        receivedRedirectURI={receivedRedirectURI}
+                        // Limit the number of times a user can submit Plaid credentials
+                        if (event === 'SUBMIT_CREDENTIALS') {
+                            App.handleRestrictedEvent(event);
+                        }
+                    }}
+                    // User prematurely exited the Plaid flow
+                    // eslint-disable-next-line react/jsx-props-no-multi-spaces
+                    onExit={onExitPlaid}
+                    receivedRedirectURI={receivedRedirectURI}
+                />
+            );
+        }
+
+        if (plaidDataErrorMessage) {
+            return <Text style={[styles.formError, styles.mh5]}>{plaidDataErrorMessage}</Text>;
+        }
+
+        if (lodashGet(plaidData, 'isLoading')) {
```

Comment:
The one thing I'm slightly worried about here... is this way, we'll never be able to show the error message AND a loading indicator at the same time, right? Where previously we could... Can we make sure this doesn't cause any regressions?

Reply:
hmm those are just optimistic loading states though, i could see a situation where an error happens and we want to try again so we set `isLoading` again, right?

Reply:
oh wait ya i see that we clear the error any time we try again so most likely they won't ever be shown at the same time